src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -24,44 +25,55 @@ public TemplateEmitter(SemanticModel model)
             this.model = model;
         }
 
+        private string CalculateTemplateHash(string assemblyVersion)
+        {
+            using MemoryStream memoryStream = new();
```

Comment:
I think you can save a bit of complexity with (not sure if any more performant):

```csharp
using (var stringWriter = new StringWriter())
using (var writer = new JsonTextWriter(stringWriter))
{
    new TemplateWriter(writer, this.model).Write(true, assemblyVersion, null);
}

return TemplateHashExtensions.ComputeTemplateHash(stringWriter.ToString());
```